src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -557,4 +567,136 @@ def calculate_sediment_deposition(
+    cdef _ManagedRaster mfd_flow_direction_raster = _ManagedRaster(
+        mfd_flow_direction_path, 1, False)
+
+    cdef _ManagedRaster average_aspect_raster = _ManagedRaster(
+        target_average_aspect_path, 1, True)
+
+    cdef int seed_row = 0
+    cdef int seed_col = 0
+    cdef int n_pixels_visited = 0
+    cdef int win_xsize, win_ysize, xoff, yoff
+    cdef int row_index, col_index, neighbor_index
+    cdef int flow_weight_in_direction
+    cdef int weight_sum
+    cdef int seed_flow_value
+    cdef float aspect_weighted_average, aspect_weighted_sum
+    cdef float proportional_flow
+    cdef float flow_length
+
+    # the flow_lengths array is the functional equivalent
+    # of calculating |sin(alpha)| + |cos(alpha)|.
+    cdef float* flow_lengths = [
+        1.0, <float>cmath.M_SQRT2,
+        1.0, <float>cmath.M_SQRT2,
+        1.0, <float>cmath.M_SQRT2,
+        1.0, <float>cmath.M_SQRT2
+    ]
+
+    # Loop over iterblocks to maintain cache locality
+    # Find each non-nodata pixel and calculate proportional flow
+    # Multiply proportional flow times the flow length x_d
+    # write the final value to the raster.
+    for offset_dict in pygeoprocessing.iterblocks(
+            (mfd_flow_direction_path, 1), offset_only=True, largest_block=0):
+        win_xsize = offset_dict['win_xsize']
+        win_ysize = offset_dict['win_ysize']
+        xoff = offset_dict['xoff']
+        yoff = offset_dict['yoff']
+
+        LOGGER.info('Average aspect %.2f%% complete', 100.0 * (
+            n_pixels_visited / float(n_cols * n_rows)))
+
+        for row_index in range(win_ysize):
+            seed_row = yoff + row_index
+            for col_index in range(win_xsize):
+                seed_col = xoff + col_index
+                seed_flow_value = <int>mfd_flow_direction_raster.get(
+                    seed_col, seed_row)
+
+                # Skip this seed if it's nodata (Currently expected to be 0).
+                # No need to set the nodata value here since we have already
+                # filled the raster with nodata values at creation time.
+                if seed_flow_value == mfd_flow_direction_nodata:
+                    continue
+
+                weight_sum = 0
+                aspect_weighted_sum = 0
+                for neighbor_index in range(8):
+                    # We only need to check the neighbors relative to the
+                    # boundaries if we're in the first or last rows and
+                    # columns.
+                    if seed_row == 0 or seed_row == n_rows:
+                        neighbor_row = seed_row + ROW_OFFSETS[neighbor_index]
+                        if neighbor_row < 0 or neighbor_row >= n_rows:
+                            continue
+
+                    if seed_col == 0 or seed_col == n_cols:
+                        neighbor_col = seed_col + ROW_OFFSETS[neighbor_index]
+                        if neighbor_col < 0 or neighbor_col >= n_cols:
+                            continue
```

Comment:
Same here, don't you need to check this even if you're not on the first or last column?